### ISSUE TITLE
CarrierWave in test does not delete file

### ISSUE THREAD

**Issue by coding-bunny** [carrierwaveuploader/carrierwave#2387]
Hello,

I'm currently upgrading CarrierWave as part of our Rails upgrade process, and noticed a change in behavior when jumpin on the master branch to resolve all deprecation warnings for Rails.

We basically had a test, where we call `delete` on a File stored with CarrierWave, and this method returns true in our testing environment.
Yet when we ask whether the file still exists, the method also returns true where this used to return false before.

Did something got changed in the way CarrierWave works in testing environments?

The test in question:

```
describe '#delete' do
   it 'deletes a file' do
     expect(s3_file).to exist
     s3_file.delete
     expect(s3_file).not_to exist
  end
end
```
This is part of our S3 Uploader spec, and it worked fine in Rails 5.0 with CarrierWave 1.2

**Comment by mshibuya** [comment]
Could you tell me the last working version of CarrierWave?

**Comment by coding-bunny** [comment]
According our Gemfile that would have been `1.0.0`

**Comment by mshibuya** [comment]
Do you run tests with transactional fixtures? CarrierWave's file operation is supposed to be run in transactional manner.
cf. #2209

**Comment by coding-bunny** [comment]
We don't run transactional fixtures.
Not sure if I can even enable that at this point in the project, or rather not sure what the consequences will be. But if that solves the problem, might be worth a try I suppose.

**Comment by coding-bunny** [comment]
I've enabled transactional fixtures, but the test still fails :

```
expected #<ShardedS3Storage::File:0x0055a7993f7730 @uploader=#<CategoryImageUploader:0x0055a799455e48 @model=#...rage_class=nil,
    encryption=nil,
    encryption_key=nil,
    version=nil,
    kms_key_id=nil
  >> not to exist

  0) ShardedS3Storage File interacting with S3 #delete deletes a file
     Failure/Error: expect(s3_file).not_to exist

       expected #<ShardedS3Storage::File:0x0055a7993f7730 @uploader=#<CategoryImageUploader:0x0055a799455e48 @model=#...rage_class=nil,
           encryption=nil,
           encryption_key=nil,
           version=nil,
           kms_key_id=nil
         >> not to exist
     # ./spec/storages/sharded_s3_storage_spec.rb:133:in `block (5 levels) in <top (required)>'
     # ./spec/rails_helper.rb:84:in `block (3 levels) in <top (required)>'
     # /home/olivar/.rvm/gems/ruby-2.3.0/gems/database_cleaner-1.7.0/lib/database_cleaner/generic/base.rb:16:in `cleaning'
     # /home/olivar/.rvm/gems/ruby-2.3.0/gems/database_cleaner-1.7.0/lib/database_cleaner/base.rb:100:in `cleaning'
     # /home/olivar/.rvm/gems/ruby-2.3.0/gems/database_cleaner-1.7.0/lib/database_cleaner/configuration.rb:86:in `block (2 levels) in cleaning'
     # /home/olivar/.rvm/gems/ruby-2.3.0/gems/database_cleaner-1.7.0/lib/database_cleaner/configuration.rb:87:in `cleaning'
     # ./spec/rails_helper.rb:83:in `block (2 levels) in <top (required)>'

17 examples, 1 failure, 16 passed

```

**Comment by mshibuya** [comment]
Can you reproduce this problem using fresh Rails app?
Uploading it to Gist or GitHub repo may help track down the issue.

**Comment by coding-bunny** [comment]
I can try setting this up when I find some free time for this...

**Comment by coding-bunny** [comment]
https://github.com/coding-bunny/carrierwave-test

All required information is inside the project's README

**Comment by mshibuya** [comment]
Ah I see, it was because of the change in https://github.com/carrierwaveuploader/carrierwave/commit/bffbaaf3b5892caa96ad875a8c1655564d17ce1c.
I've just pushed a fix as https://github.com/carrierwaveuploader/carrierwave/commit/f94b0615c1b0c1000c31cd64b373dae85f0f4893.

Thanks for reporting!

**Comment by coding-bunny** [comment]
Thank you for fixing it :D

**Comment by coding-bunny** [comment]
minor side question : Will a separate release go out for this "soonish", we're trying to avoid pointing to master branch with gems.

**Comment by mshibuya** [comment]
The fix will be included in upcoming release of 2.0.0, which I hope will be in June or so...

**Comment by mriddle** [comment]
This issue was fixed in 2.0.0 (f94b0615c1b0c1000c31cd64b373dae85f0f4893) but remains in 1.3.2 and should be backported if there's a plan to continue support for the 1.x series.
